ui/src/components/data-source-panel.tsx

```diff
@@ -116,24 +123,32 @@ export default class DataSourcePanel extends React.Component<DataSourcePanelProp
 
    render() {
       return (
-         <div className="relative flex flex-col bg-[#221f2e] items-start px-8 pt-0 pb-4 min-h-[300px]">
+         <div className="relative flex flex-col bg-[#221f2e] items-start px-8 pt-0 pb-4 min-h-[300px]">1
             {
                !this.state.isAdding && <h1 className="mt-4 relative self-center text-white block text-4xl mb-8 font-poppins">Data Source Panel</h1>
             }
-            <IoMdClose onClick={this.props.onClose} className='absolute right-4 top-3 text-2xl text-white hover:text-[#9875d4] hover:cursor-pointer'></IoMdClose>
+
+            {/* X and Edit in top right */}
+            <div className="absolute flex flex-col items-center right-4 top-3 text-2xl text-white gap-4">
```

Comment:
Move the edit button to be in the same line with the title, and make it bigger + a little margin right
![image](https://user-images.githubusercontent.com/127092381/228069133-6dfbd71b-464f-42be-a90f-e3ba55e6de71.png)